cmake/kokkos_tpls.cmake

```diff
@@ -15,6 +15,10 @@ KOKKOS_TPL_OPTION(CUDA    Off)
 KOKKOS_TPL_OPTION(LIBRT   Off)
 KOKKOS_TPL_OPTION(LIBDL   On)
 
+IF(KOKKOS_ENABLE_PROFILING AND NOT KOKKOS_ENABLE_LIBDL)
+  MESSAGE(SEND_ERROR "Kokkos_ENABLE_PROFILING requires Kokkos_ENABLE_LIBDL=ON")
```

Comment:
Did you consider emitting a warning and disabling the profiling hooks if libdl is not found?

Reply:
I thought about changing the default value, but this file is parsed after the one setting up the profiling. 
I rather have a configure time error than overriding what the user requests (even emitting a warning).

Reply:
I am not 100% sold on that behavior but I won't object if we issue a better error message.

Reply:
What don't you like about the error message? In which regard could it be improved?

Reply:
On second thought, this is fine.  I didn't realize the current behavior is to error out if libdl is not found, even for the case where the user did not explicitly enabled it.